### PR TITLE
Add webpack watcher on partial include/extends/embed 

### PR DESCRIPTION
**Pull request by ryuran**
- add editor config
- add partial watcher (addDependency)
- move `var` to `const` and `let`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -156,14 +159,16 @@ function customRenderFn(renderFn) {
...
  * @param {any} engine
  * @param {string} str
  * @param {Object} locals
  * @returns {string}
  */
 function render(engine, str, locals, engineOptions) {
+  let output = null;
   try {
-    var output = engine.render(engine.engine, str, locals, engineOptions)
+    output = engine.render.call(this, engine.engine, str, locals, engineOptions)
```
Comment (jabney) [inline index.js]: Add a comment here about the identity of `this` in this context.

File: lib/renderers.js [hunk 2]
```diff
@@ -19,13 +19,26 @@ function mustache(engine, str, locals, options) {
...
+  engine.extend((Twig) => {
+    const defaultSave = Object.assign(Twig.Templates.save);
+    Twig.Templates.save = function customSave(template) {
+      registry.push(template.path);
+      return defaultSave.call(this, template);
+    };
+  });
+  const template = engine.twig(opt)
+  registry.forEach(this.addDependency);
+
```
Comment (jabney) [inline lib/renderers.js]: Is there a straightforward way to [write a test](https://github.com/jabney/render-template-loader/blob/master/test/twig.spec.js) for this functionality?

Comment (ryuran) [reply lib/renderers.js]: I currently don't know how to write this test but I agréé, de should add it.

File: index.js (hunk 1 above)
Comment (ryuran) [reply index.js]: https://github.com/jabney/render-template-loader/pull/6/files#diff-168726dbe96b3ce427e7fedce31bb0bcR162 is not enough ?

**Comment by ryuran** [comment]
@jabney ready to review.
I saw a mistake when i was trying to add test. It's fixed now.

File: index.js (hunk 1 above)
Comment (jabney) [reply index.js]: That works!

**Comment by jabney** [comment]
[render-template-loader v1.2.3 on npm](https://www.npmjs.com/package/render-template-loader)

**Comment by jabney** [comment]
@ryuran, 

If you want, submit a pull request with your contributor info in the `package.json` file and I'll merge it in. Use any or all of these keys:

`package.json`:

```javascript
"contributors": [
  {
    "name": "...",
    "email": "...",
    "url": "http://..."
  }
]
```

Or just reply here with your info and I'll add it myself.

**Pull request merged**